From ae04d10c69337940e569c8b6930539dde1e5562a Mon Sep 17 00:00:00 2001
From: Kafva <kafva.one@gmail.com>
Date: Sat, 24 Aug 2024 18:02:56 +0200
Subject: [PATCH 1/1] Use lua builtin sort()

This allows us to use fidget notifications inside the callbacks of
vim.system()
---
 lua/fidget/notification/model.lua | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lua/fidget/notification/model.lua b/lua/fidget/notification/model.lua
index 19fa42e..452170d 100644
--- a/lua/fidget/notification/model.lua
+++ b/lua/fidget/notification/model.lua
@@ -267,7 +267,7 @@ function M.update(now, configs, state, msg, level, opts)
   if new_index then
     -- NOTE: we use vim.fn.sort() here because it is stable.
     -- :h sort() docs claim that it does so in-place, but it doesn't.
-    state.groups = vim.fn.sort(state.groups, function(a, b)
+    table.sort(state.groups or {}, function(a, b)
       return (a.config.priority or 50) - (b.config.priority or 50)
     end)
   end
-- 
2.46.0

